courses/views/v1/__init__.py

```diff
@@ -240,3 +248,47 @@ def partial_update(self, request, *args, **kwargs):
             # to update the rest of the fields in the PATCH request
             # or separate out the APIs into function-based views.
             raise NotImplementedError
+
+
+@api_view()
+@permission_classes([IsAuthenticated])
+def get_user_program_enrollments(request):
+    """
+    Returns a unified set of program and course enrollments for the current
+    user.
+    """
+
+    courseruns = (
+        CourseRunEnrollment.objects.filter(user=request.user)
+        .select_related("run__course__page")
+        .select_related("run__course__program")
+        .all()
+    )
+
+    program_list = {}
+
+    for enrollment in courseruns:
+        if enrollment.run.course.program is not None:
+            if enrollment.run.course.program.id in program_list:
+                program_list[enrollment.run.course.program.id]["enrollments"].append(
+                    enrollment
+                )
+            else:
+                program_list[enrollment.run.course.program.id] = {
+                    "enrollments": [enrollment],
+                    "program": enrollment.run.course.program,
+                }
```

Comment:
Just so I understand, this is working like: "if the program ID is already in the list, then add the course enrollment for each course the user is enrolled in under that program, otherwise add the the program ID to the list?

Reply:
Yup, just using the keys in the dict to dedup the program enrollment list.